Assets/Xbox Live/Scripts/SignInManager.cs

```diff
@@ -0,0 +1,605 @@
+                playerInfo.SignOutCallbacks.Remove(callback);
+            }
+        }
+    }
+
+    /// <summary>
+    /// Returns a list of the currently signed in Xbox Live Users and their assigned player numbers
+    /// </summary>
+    /// <returns>A <see cref="Dictionary{TKey, TValue}"/> with player number keys and Xbox Live User values.</returns>
+    public Dictionary<int, XboxLiveUser> GetCurrentPlayers()
+    {
+        var currentPlayerDictionary = new Dictionary<int, XboxLiveUser>();
+        foreach (var playerInfo in this.CurrentPlayers)
+        {
+            if (playerInfo.Value.XboxLiveUser != null)
+            {
+                currentPlayerDictionary.Add(playerInfo.Key, playerInfo.Value.XboxLiveUser);
+            }
+        }
+        return currentPlayerDictionary;
+    }
+
+    /// <summary>
+    /// Returns the number of signed in Xbox Live Users.
+    /// </summary>
+    /// <returns>An integer representing the number of currently signed in players.</returns>
+    public int GetCurrentNumberOfPlayers()
+    {
+        return this.GetCurrentPlayers().Count;
+    }
+
+    /// <summary>
+    /// Returns the maximum number of signed in Xbox Live Users supported on current platform.
+    /// </summary>
+    /// <returns>An integer representing the maximum number of players supported on the current platform.</returns>
+    public int GetMaximumNumberOfPlayers()
+    {
+        if (Application.isEditor)
+        {
+            return 16;
+        }
+        else
+        {
+            if (SystemInfo.deviceType == DeviceType.Console)
+            {
+                return 16;
+            }
+            else
+            {
+                return 1;
+            }
+        }
+    }
+
+    /// <summary>
+    /// Returns the player number assigned to the Xbox Live User with input xuid.
+    /// </summary>
+    /// <param name="xuid">The xuid of the Xbox Live User</param>
+    /// <returns>The player number assigned to the Xbox Live User or -1 if no Xbox Live Users are signed in with that xuid.</returns>
+    public int GetPlayerNumber(string xuid)
+    {
+        foreach (var playerInfo in this.CurrentPlayers)
+        {
+            if (playerInfo.Value.XboxLiveUser != null && playerInfo.Value.XboxLiveUser.XboxUserId.Equals(xuid))
+            {
+                return playerInfo.Key;
+            }
+        }
+        return -1;
+    }
+
+    /// <summary>
+    /// Returns the Xbox Live Context for the Xbox Live User with the assigned player number.
+    /// </summary>
+    /// <param name="playerNumber">The player number.</param>
+    /// <returns>The Xbox Live Context for the Xbox Live User with the assigned player number.</returns>
+    public XboxLiveContext GetXboxLiveContextForPlayer(int playerNumber)
+    {
+        if (ValidatePlayerNumber(playerNumber, "Get Xbox Live Context", XboxLiveOperationType.GetUser))
+        {
+            return new XboxLiveContext(this.CurrentPlayers[playerNumber].XboxLiveUser);
```

Comment:
>return new XboxLiveContext(this.CurrentPlayers[playerNumber].XboxLiveUser); [](start = 12, length = 75)

Maybe we should cache the XboxLiveContext as there should only be one for the lifetime of a user

Reply:
Great point. Fixed. Will be in the next push.